sdk/cosmos/azure-cosmos/azure/cosmos/diagnostics.py

```diff
@@ -22,8 +22,6 @@
 """Diagnostic tools for Azure Cosmos database service operations.
 """
 
-from requests.structures import CaseInsensitiveDict
```

Comment:
from azure.core.utils import CaseInsensitiveDict

Reply:
this breaks the tests due to being unable to import it:
![image](https://github.com/Azure/azure-sdk-for-python/assets/30335873/dd260f3f-f756-48b3-ae85-bc617febc956)

Reply:
What your currently supported min Azure Core version? Maybe we need to bump it.